src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -685,9 +688,13 @@ private void unsubscribeNodeEvents(NodeViewModel nodeViewModel)
 
         void Model_NodeRemoved(NodeModel node)
         {
-            NodeViewModel nodeViewModel = Nodes.First(x => x.NodeLogic == node);
-            Errors.Remove(nodeViewModel.ErrorBubble);
-            Nodes.Remove(nodeViewModel);
+            NodeViewModel nodeViewModel;
+            lock (Nodes)
+            {
+                nodeViewModel = Nodes.First(x => x.NodeLogic == node);
```

Comment:
I guess this is safest, but is there any harm in scanning the collection outside the lock?

Reply:
The thing is you can run into errors by enumerating over it while it's being modified. That's the initial crash I found.